### PR TITLE
feat: add dialtone health check script

### PR DESCRIPTION
**Pull request by braddialpad**
## Description
Adds a script which consumers can use to check dialtone health in their app. It does so by looking through specified files for CSS properties that are not set to dialtone values. Could be expanded to do much more cool stuff in the future but this is a good start. See comments and usage info in the code itself for more detailed doc.

Example output

You'll notice there are some false positives with hsla() and such, will iterate on this further to see if I can fix those, for now don't think its that huge a deal since it's easily ignored.

```
node ./find_non_dialtone_properties.js ./js/components
        non-dialtone property found on line 269:   background-color: #FFFFFF;
        non-dialtone property found on line 305:   background-color: #5078EE;
        non-dialtone property found on line 320:   color: #000000;
js/components/app_settings/custom_ringtone.vue: 3
        non-dialtone property found on line 115:   background-color: rgb(127, 150, 157);
        non-dialtone property found on line 124:   background-color: rgba(0, 0, 0, 0.5);
        non-dialtone property found on line 132:   color: white;
js/components/base_avatar_list.vue: 3
        non-dialtone property found on line 156:   color: inherit;
js/components/base_call_recording_controls_actions.vue: 1
        non-dialtone property found on line 125:     background-color: transparent;
        non-dialtone property found on line 130:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2);
        non-dialtone property found on line 134:     background-color: transparent;
        non-dialtone property found on line 139:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2) !important;
        non-dialtone property found on line 156:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2) !important;
        non-dialtone property found on line 160:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2) !important;
        non-dialtone property found on line 169:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2);
        non-dialtone property found on line 173:     background-color: transparent;
        non-dialtone property found on line 219:     border-color: transparent;
        non-dialtone property found on line 233:   background-color: white !important;
        non-dialtone property found on line 254:   color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.8) !important;
        non-dialtone property found on line 255:   border-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.6) !important;
        non-dialtone property found on line 264:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2) !important;
        non-dialtone property found on line 268:   background-color: hsla(var(--primary-color-h), var(--primary-color-s), var(--primary-color-l), 0.2) !important;
        non-dialtone property found on line 292:     background-color: transparent !important;
js/components/base_date_picker.vue: 15
        non-dialtone property found on line 206:   background-color: #888888;
js/components/base_media_controls.vue: 1
        non-dialtone property found on line 150:       background-color: transparent;
js/components/base_status_icon.vue: 1
        non-dialtone property found on line 25:   color: #ffffff;
js/components/base_text_tooltip.vue: 1
        non-dialtone property found on line 1160:     background-color: hsla(~"@{primary-color--hsl} / 10%");
js/components/call_review/call_review.vue: 1
        non-dialtone property found on line 382:     background-color: hsla(~"@{primary-color--hsl} / 10%");
        non-dialtone property found on line 406:   background-color: transparent;
js/components/call_review/call_review_excerpts_section_item.vue: 2
        non-dialtone property found on line 115:   color: white;
        non-dialtone property found on line 130:     color: black;
js/components/call_review/call_review_header/call_review_header_avatars.vue: 2
        non-dialtone property found on line 176:   background-color: hsla(~"@{primary-color--hsl} / 10%");
        non-dialtone property found on line 183:     background-color: hsla(~"@{primary-color--hsl} / 13%") !important;
        non-dialtone property found on line 198:     color: transparent;
        non-dialtone property found on line 212:   border-color: transparent !important;
js/components/call_review/call_review_header/call_review_header_search.vue: 4
        non-dialtone property found on line 73:     background-color: hsla(~"@{primary-color--hsl} / 10%");
js/components/call_review/call_review_moment_nav/call_review_moment_nav_category.vue: 1
        non-dialtone property found on line 195:     color: darken(@purple-300, 15%);
js/components/call_review/call_review_moment_nav/call_review_moment_nav_header.vue: 1
        non-dialtone property found on line 121:     background-color: hsla(~"@{primary-color--hsl} / 10%");
js/components/call_review/call_review_row_controls_wrapper.vue: 1
        non-dialtone property found on line 259:   background-color: hsla(~"@{primary-color--hsl} / 10%");
        non-dialtone property found on line 266:     background-color: hsla(~"@{primary-color--hsl} / 13%") !important;
js/components/call_review/call_review_search_controls.vue: 2
        non-dialtone property found on line 463:     background-color: hsl(var(--primary-color-h), var(--primary-color-s), calc(var(--primary-color-l) + 30%));
js/components/call_snippet_recording_controls.vue: 1
        non-dialtone property found on line 91:   background-color: #F4F4F4;
js/components/call_summary/call_summary_arrows.vue: 1
        non-dialtone property found on line 80:   color: #5078EE;
        non-dialtone property found on line 86:   color: #5078EE;
js/components/call_summary/call_summary_participant_list.vue: 2
        non-dialtone property found on line 523:   background-color: #5078ee;
        non-dialtone property found on line 534:   color: #ffffff;
        non-dialtone property found on line 542:   color: #ffffff;
        non-dialtone property found on line 545:   background-color: rgba(34, 34, 34, 0.8);
        non-dialtone property found on line 560:   color: #444444;
        non-dialtone property found on line 572:   color: #444444;
        non-dialtone property found on line 583:   background-color: rgba(255, 255, 255, 0.85);
js/components/call_summary/call_summary_transcript.vue: 7
        non-dialtone property found on line 249:   background-color: transparent;
js/components/call_summary/call_summary_transcript_header.vue: 1
        non-dialtone property found on line 57:     background-color: transparent;
        non-dialtone property found on line 68:     color: #222528; /* Dark Grey 1 */
js/components/dialog_header.vue: 2
        non-dialtone property found on line 160:   background-color: transparent;
        non-dialtone property found on line 180:   background-color: transparent;
        non-dialtone property found on line 181:   border-color: transparent !important;
        non-dialtone property found on line 203:   background-color: transparent;
        non-dialtone property found on line 204:   border-color: transparent !important;
        non-dialtone property found on line 226:   background-color: transparent;
        non-dialtone property found on line 227:   border-color: transparent !important;
        non-dialtone property found on line 230:   color: transparent;
        non-dialtone property found on line 235:   background-color: transparent;
js/components/double_range_slider.vue: 9
        non-dialtone property found on line 396:   background-color: #EEE;
        non-dialtone property found on line 411:   background-color: hsla(~"@{primary-color--hsl} / 10%");
        non-dialtone property found on line 424:   background-color: transparent;
js/components/dropdown_menu.vue: 3
        non-dialtone property found on line 98:     color: #222222;
        non-dialtone property found on line 123:     background-color: #DCEEFF;
        non-dialtone property found on line 138:     color: #222222;
js/components/dropdown_section.vue: 3
        non-dialtone property found on line 49:     background-color: #FFF;
        non-dialtone property found on line 53:     background-color: #FFF;
js/components/feed_item_dropdown_option.vue: 2
        non-dialtone property found on line 173:   background-color: fade(@black-900, 95%);
        non-dialtone property found on line 184:   border-color: transparent transparent fade(@black-900, 95%) transparent;
js/components/feedback.vue: 2
        non-dialtone property found on line 82:     border-color: #95928E;
        non-dialtone property found on line 91:       background-color: #FBFAF8 !important;
        non-dialtone property found on line 95:       background-color: #FBFAF8 !important;
js/components/guest/third_party_auth_button.vue: 3
        non-dialtone property found on line 82:       background-color: #FBFAF8 !important;
        non-dialtone property found on line 86:       background-color: #DCE4F1 !important;
        non-dialtone property found on line 89:     background-color: #EFE4E0 !important;
js/components/guest/third_party_button.vue: 3
        non-dialtone property found on line 178:   color: #ffffff;
js/components/incoming_call_header.vue: 1
        non-dialtone property found on line 26:     color: #696969;
js/components/login/login_divider.vue: 1
        non-dialtone property found on line 135:   border-color: #f4f4f4;
        non-dialtone property found on line 143:   background-color: #dceeff;
        non-dialtone property found on line 153:   background-color: rgba(221,221,221,0.5);
        non-dialtone property found on line 159:   background-color: #DDDDDD;
        non-dialtone property found on line 175:   color: #7d7d7f;
        non-dialtone property found on line 179:   color: #bbbbbb;
        non-dialtone property found on line 191:   background-color: rgba(244,244,244,0.5);
        non-dialtone property found on line 200:   background-color: rgba(244,244,244,0.5);
        non-dialtone property found on line 205:   color: #222222;
        non-dialtone property found on line 209:   background-color: #dceeff;
js/components/moment_filters.vue: 10
        non-dialtone property found on line 26:   background-color: #f4f4f4;
js/components/rightbar_footer.vue: 1
        non-dialtone property found on line 99:   border-color: transparent !important;
        non-dialtone property found on line 120:     background-color: hsla(var(--black-900-h) var(--black-900-s) var(--black-900-l) / var(--bgo)) !important;
js/components/screen_recording_playback_thumbnails.vue: 2
        non-dialtone property found on line 161:     color: #222;
        non-dialtone property found on line 187:     background-color: #eee;
        non-dialtone property found on line 191:     background-color: hsla(~"@{primary-color--hsl} / 10%");
js/components/select_menu_option.vue: 3
        non-dialtone property found on line 322:       font-family: inherit;
        non-dialtone property found on line 323:       color: initial;
        non-dialtone property found on line 340:     border-color: transparent;
js/components/select_menu_value.vue: 3
        non-dialtone property found on line 194:   background-color: rgba(221,221,221,0.8);
        non-dialtone property found on line 199:   background-color: #f4f4f4;
        non-dialtone property found on line 210:   color: white;
        non-dialtone property found on line 233:   background-color: #7d7d7d;
        non-dialtone property found on line 234:   color: white;
        non-dialtone property found on line 246:       background-color: hsla(var(--white-h) var(--white-s) var(--white-l) / var(--bgo));
        non-dialtone property found on line 259:   background-color: #5078ee;
        non-dialtone property found on line 265:   background-color: rgba(255, 255, 255, 0.7);
        non-dialtone property found on line 271:   background-color: rgba(255, 255, 255, 0.7);
js/components/transcript_text_bubble.vue: 9
        non-dialtone property found on line 513:   background-color: #eceff9;
        non-dialtone property found on line 518:   background-color: #DDE5FC;
        non-dialtone property found on line 548:   color: #222222;
        non-dialtone property found on line 555:   color: #888888;
js/components/transcript_text_row.vue: 4
        non-dialtone property found on line 100:     color: #5078EE;
js/components/whats_new/whats_new.vue: 1
        non-dialtone property found on line 94:     background-color: rgba(244,244,244,0.7);
        non-dialtone property found on line 103:     background-color: #18D4D4;
        non-dialtone property found on line 120:     color: #5078EE;
        non-dialtone property found on line 128:     color: #222222;
        non-dialtone property found on line 149:     color: #222222;
        non-dialtone property found on line 157:     color: #888888;
js/components/whats_new/whats_new_feed_row.vue: 6
Found 119 total color, background-color, border-color, font-family    CSS properties that are not dialtone in this directory.
```

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: lib/build/js/dialtone_health_check.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env node
```
Comment (josedialpad) [inline lib/build/js/dialtone_health_check.js:1]: chmod +x? I couldn't execute using `npm link` without making it executable first.

File: lib/build/js/dialtone_health_check.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  'font-family',
+];
+
+const extensionFilters = options['--ext']?.split(',') ?? [''];
+let properties = options['--property']?.split(',');
+if (!properties) properties = defaultSearch;
+
+function fromDir (startPath, filter) {
+  if (!fs.existsSync(startPath)) {
+    console.log('no dir ', startPath);
```
Comment (josedialpad) [inline lib/build/js/dialtone_health_check.js]:
```suggestion
    console.error('Directory not found'); exit(1);
```

File: lib/build/js/dialtone_health_check.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  'background-color',
+  'border-color',
+  'font-family',
+];
+
+const extensionFilters = options['--ext']?.split(',') ?? [''];
+let properties = options['--property']?.split(',');
+if (!properties) properties = defaultSearch;
+
+function fromDir (startPath, filter) {
```
Comment (josedialpad) [inline lib/build/js/dialtone_health_check.js]: nit: is the extension the only filter? if so maybe just calling `fileExtension` is clearer?

File: lib/build/js/dialtone_health_check.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  return results;
+}
+const files = [];
+extensionFilters.forEach(extension => {
+  files.push(...fromDir(options['<path>'], extension));
+});
+if (files.length <= 0) { console.error('No files found in directory'); exit(1); }
+
+const regex = properties.map(property => {
+  return new RegExp(`^\\s*${property}:\\s(?:(?!var\\(--))(?:(?!@)).*`, 'gm');
```
Comment (josedialpad) [inline lib/build/js/dialtone_health_check.js:70]: those custom CSS properties or LESS vars defined by the user won't be caught here? Not sure if there are many of these cases in the product but if there are, maybe it's significant to try to catch those cases?

**Review by josedialpad — COMMENTED** [review]
Tried in ubervoice in a couple of dirs and options and works like a charm!

File: lib/build/js/dialtone_health_check.js (hunk 1 above)
Comment (braddialpad) [reply lib/build/js/dialtone_health_check.js:1]: ah yes, thanks

File: lib/build/js/dialtone_health_check.js (hunk 3 above)
Comment (braddialpad) [reply lib/build/js/dialtone_health_check.js]: Yeah I agree

File: lib/build/js/dialtone_health_check.js (hunk 4 above)
Comment (braddialpad) [reply lib/build/js/dialtone_health_check.js:70]: Yeah unfortunately that would be near impossible to do with pure regex. Might be a future update. Could possibly compare the variable name to the entire list of variables in the actual dialtone package, or the "most proper" way walking the CSS AST tree to find if it was locally declared, but that's way outside the scope of what we're trying to do in this PR.

**Comment by braddialpad** [comment]
Requested changes are in

File: lib/build/js/dialtone_health_check.js (hunk 4 above)
Comment (josedialpad) [reply lib/build/js/dialtone_health_check.js:70]: Yeah, it can be nice to have but definitely tricky to get

**Review by josedialpad — APPROVED** [review]
Great work! I think could be good to send a report and get feedback about it

**Pull request merged**